docs/spec/concepts.rst

```diff
@@ -18,17 +348,44 @@ Example::
           e = [e]
       ...
 
-A type factored by ``T1 | T2 | ...`` is a supertype
-of all types ``T1``, ``T2``, etc., so that a value that
-is a member of one of these types is acceptable for an argument
-annotated by ``T1 | T2 | ...``.
+A fully static union type ``T1 | T2``, where ``T1`` and ``T2`` are fully static
+types, represents the set of values formed by the union of the sets of values
+represented by ``T1`` and ``T2``, respectively. Thus, by the definition of the
+supertype relation, the union ``T1 | T2`` is a supertype of both ``T1`` and
+``T2``, and ``T1`` and ``T2`` are both subtypes of ``T1 | T2``.
 
-One common case of union types are *optional* types.  By default,
-``None`` is an invalid value for any type, unless a default value of
-``None`` has been provided in the function definition.  Examples::
+A gradual union type ``S1 | S2``, where ``S1`` and ``S2`` are gradual types,
```

Comment:
```suggestion
A gradual union type ``S1 | S2``, where ``S1`` or ``S2`` are gradual types,
```
is this right? if a fully static type is a gradual type, we should mention that when defining gradual type

Reply:
I think if you read Carl's PR closely, it's clear that all fully static types are gradual types: all Python types are gradual, so the set of fully static types is a strict subset of the set of gradual types (which is the same as the set of all types). This is indicated in passages in `concepts.rst` such as:

>     A **gradual type** can be a fully static type, :ref:`Any` itself, or a type
>     that contains a gradual form as a sub-part.

and in passages in `glossary.rst` such as:

>      Types in the Python type system are "gradual". A gradual type may be a
>      :term:`fully static type`, or it may be :ref:`Any`, or a type that
>      contains ``Any`` or another :term:`gradual form`.

Nonetheless, I agree that this is quite a subtle point and one that I was confused about in a prior version of the PR (see https://github.com/python/typing/pull/1743#discussion_r1623286059). I think it's easy to misread as although the language of the document is clear, the structure of the document overall often pits the two concepts against each other in contrast (first talking about one category, then talking about how it's different for the other category), such that a reader can easily misunderstand the two categories as being mutually exclusive. It's also not helped by the fact that we don't seem to have a precise term for "gradual types that include at least one gradual form as a subelement, such that the materialization from gradual type to fully static type transforms the type in some way".

Here are some possible edits to the PR that might clarify this somewhat for readers:

```diff
--- a/docs/spec/concepts.rst
+++ b/docs/spec/concepts.rst
@@ -54,6 +54,11 @@ as **fully static types**.
 A **gradual type** can be a fully static type, :ref:`Any` itself, or a type
 that contains a gradual form as a sub-part.
 
+The categories "fully static" and "gradual" are not mutually exclusive.
+Python's type system is gradual; all Python types are gradual types, including
+those types which are fully static. The set of fully static types is a strict
+subset of the set of gradual types.
+
 Fully static types
 ~~~~~~~~~~~~~~~~~~
 
@@ -70,11 +75,11 @@ If an object ``v`` is a member of the set of objects denoted by a fully static
 type ``T``, we can say that ``v`` is a "member of" the type ``T``, or ``v``
 "inhabits" ``T``.
 
-Gradual types
-~~~~~~~~~~~~~
+Gradual forms and gradual types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-:ref:`Any` represents an unknown static type. It denotes some unknown set of
-runtime values.
+:ref:`Any` is a gradual form that represents an unknown static type. It denotes
+some unknown set of runtime values.
 
 This may appear similar to the fully static type ``object``, which represents
 the set of all Python objects, but it is quite different.
diff --git a/docs/spec/glossary.rst b/docs/spec/glossary.rst
index 41bb2b5..1e9bae2 100644
--- a/docs/spec/glossary.rst
+++ b/docs/spec/glossary.rst
@@ -55,8 +55,9 @@ This section defines a few terms that may be used elsewhere in the specification
    fully static type
       A type is "fully static" if it does not contain any :term:`gradual form`.
       Fully static types represent a set of possible runtime values. Fully
-      static types participate in the :term:`subtype` relation. See
-      :ref:`type-system-concepts`.
+      static types participate in the :term:`subtype` relation. Just because a
+      type is fully static does not mean that it is not gradual: all types in
+      Python's type system are gradual types. See :ref:`type-system-concepts`.
 
    gradual form
       A gradual form is a :term:`type expression` which makes the type it is
@@ -68,7 +69,7 @@ This section defines a few terms that may be used elsewhere in the specification
       other :ref:`tuple <tuples>` types).
 
    gradual type
-      Types in the Python type system are "gradual". A gradual type may be a
+      All types in the Python type system are "gradual". A gradual type may be a
       :term:`fully static type`, or it may be :ref:`Any`, or a type that
       contains ``Any`` or another :term:`gradual form`. A gradual type does not
       necessarily represent a single set of possible runtime values; instead it
```

Reply:
Ah thanks, I'd missed that. When I tried to find an answer, I went and looked at the "Gradual types" section, which doesn't mention this (although the "Fully static and gradual types" supersection does).

I think your changes suggested changes would have avoided my confusion, although the "Gradual types" section still has some wording that implies "gradual type" as a shorthand for "type containing a gradual form"